src/watcher.py

```
@@ -0,0 +1,81 @@
+    def __init__(self, db: ChatDBClient, cache: InMemoryCache, loop: asyncio.AbstractEventLoop):
+        self.db   = db
+        self.cache = cache
+        self.loop = loop
+        self._lock = asyncio.Lock()
+    def on_modified(self, event):
+        if event.is_directory:
+            return
+        if Path(str(event.src_path)).name not in ("chat.db", "chat.db-wal"):
+            return
+        self.loop.call_soon_threadsafe(asyncio.create_task, self.handle())
+    async def handle(self):
```

Flake8 (7.2.0)

[error] expected 1 blank line, found 0

(E301)